pow/src/miner.rs

```diff
@@ -77,31 +62,33 @@ pub struct MinerBuilder {
 }
 
 impl MinerBuilder {
-    /// Sets the desired number of workers for the `Miner` nonce provider.
+    /// Create a new `MinerBuilder.
+    pub fn new() -> Self {
+        Self { ..Default::default() }
+    }
+
+    /// Sets the desired number of workers for the [`Miner`].
     pub fn with_num_workers(mut self, num_workers: usize) -> Self {
         self.num_workers.replace(num_workers);
         self
     }
 
-    /// Sets a `MinerCancel to abort the `Miner` nonce provider.
+    /// Sets a `MinerCancel to abort the [`Miner`].
     pub fn with_cancel(mut self, cancel: MinerCancel) -> Self {
         self.cancel.replace(cancel);
         self
     }
-}
-
-impl NonceProviderBuilder for MinerBuilder {
-    type Provider = Miner;
 
-    fn finish(self) -> Miner {
+    /// Build the [`Miner`].
+    pub fn finish(self) -> Miner {
         Miner {
             num_workers: self.num_workers.unwrap_or(DEFAULT_NUM_WORKERS),
             cancel: self.cancel.unwrap_or_else(MinerCancel::new),
         }
     }
 }
 
-/// A nonce provider that mine nonces.
+/// A nonce miner
```

Comment:
The trait was causing more issues than solving and it wasn't really playing nice with the client. There is always one of these methods at a given time since they are for wasm or not wasm, so the trait is not really needed for one method.